libraries/AP_RCTelemetry/AP_VideoTX.cpp

```diff
@@ -0,0 +1,247 @@
+}
+
+bool AP_VideoTX::get_band_and_channel(uint16_t freq, VideoBand& band, uint8_t& channel)
+{
+    for (int i = 0; i < AP_VideoTX::MAX_BANDS; i++) {
+        for (int j = 0; j < VTX_MAX_CHANNELS; j++) {
+            if (VIDEO_CHANNELS[i][j] == freq) {
+                band = VideoBand(i);
+                channel = j;
+                return true;
+            }
+        }
+    }
+    return false;
+}
+
+// set the current power
+void AP_VideoTX::set_configured_power_mw(uint16_t power, bool force) {
+    if (!_power_mw.configured_in_storage() || force) {
+        _power_mw.set_and_save(power);
+    }
+}
+
+// set the power in dbm, rounding appropriately
+void AP_VideoTX::set_power_dbm(uint8_t power) {
+    switch (power) {
+    case 14:
+        _current_power = 25;
+        break;
+    case 20:
+        _current_power = 100;
+        break;
+    case 23:
+        _current_power = 200;
+        break;
+    case 26:
+        _current_power = 400;
+        break;
+    case 27:
+        _current_power = 500;
+        break;
+    case 29:
+        _current_power = 800;
+        break;
+    default:
+        _current_power = uint16_t(roundf(powf(10, power / 10.0f)));
+        break;
+    }
+}
+
+// get the power in dbm, rounding appropriately
+uint8_t AP_VideoTX::get_configured_power_dbm() const {
+    switch (_power_mw.get()) {
+    case 25:    return 14;
+    case 100:   return 20;
+    case 200:   return 23;
+    case 400:   return 26;
+    case 500:   return 27;
+    case 800:   return 29;
+    default:
+        return uint8_t(roundf(10.0f * log10f(_power_mw)));
+    }
+}
+
+// set the current band
+void AP_VideoTX::set_configured_band(uint8_t band, bool force) {
+    if (!_band.configured_in_storage() || force) {
+        _band.set_and_save(band);
+    }
+}
+
+// set the current channel
+void AP_VideoTX::set_configured_channel(uint8_t channel, bool force) {
+    if (!_channel.configured_in_storage() || force) {
+        _channel.set_and_save(channel);
+    }
+}
+
+// set the current channel
+void AP_VideoTX::set_enabled(bool enabled) {
+    _current_enabled = enabled;
+    if (!_enabled.configured_in_storage()) {
+        _enabled.set_and_save(enabled);
+    }
+}
+
+// set the current options
+void AP_VideoTX::set_configured_options(uint8_t options, bool force) {
+    if (!_options.configured_in_storage() || force) {
+        _options.set_and_save(_options);
+    }
+}
+
+// peiodic update
+void AP_VideoTX::update(void)
+{
+#if HAL_CRSF_TELEM_ENABLED
+    AP_CRSF_Telem* crsf = AP::crsf_telem();
+
+    if (crsf != nullptr) {
+        crsf->update();
+    }
+#endif
+}
+
+bool AP_VideoTX::have_params_changed() const
+{
+    return update_power()
+        || update_band()
+        || update_channel()
+        || update_options();
+}
+
+// set the current configured values if not currently set in storage
+// this is necessary so that the current settings can be seen
+void AP_VideoTX::set_defaults()
+{
+    if (!_options.configured()) {
```

Comment:
configured is quite an expensive option as it involves an eeprom scan. Why do we need to know if configured?

Reply:
The code operates in two modes - first you don't configure any VTX optinos and the VTX params get populated from the VTX frames so that you can see what they are. Second you do configure the VTX options, do an update and then wait for the VTX frames to come back to confirm that the update happened. This code is for the first of these options. VTX frames happen once every 10 seconds or so, so low cost - but I could make sure that this only happens when disarmed

Reply:
Ok, I have made this happen only once and we output a friendly message about the state